### PR TITLE
feat: add allowAdditionalProperties option to rules

### PR DESCRIPTION
**Pull request by IgorKarpiuk**
## What/Why/How?

- Add allowAdditionalProperties to built in rules
- Mark disallowAdditionalProperties as deprecated

## Reference

Close: https://github.com/Redocly/redocly-cli/issues/793

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/rules/common/no-invalid-parameter-examples.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { UserContext } from '../../walk';
 import { Oas3Parameter } from '../../typings/openapi';
-import { validateExample } from '../utils';
+import { getAdditionalPropertiesOption, validateExample } from '../utils';
 
 export const NoInvalidParameterExamples: any = (opts: any) => {
-  const disallowAdditionalProperties = opts.disallowAdditionalProperties ?? true;
+  let allowAdditionalProperties = getAdditionalPropertiesOption(opts) ?? false;
```
Comment (tatomyr) [inline packages/core/src/rules/common/no-invalid-parameter-examples.ts:6]: Why do you need `let` here?

File: packages/core/src/config/minimal.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export default {
   oas3_0Rules: {
```
Comment (SmoliyY) [inline packages/core/src/config/minimal.ts:36]: Please take a look, also on `oas3_1Rules` set, maybe we want to add this possibility there too?

File: packages/core/src/rules/common/no-invalid-parameter-examples.ts (hunk 1 above)
Comment (IgorKarpiuk) [reply packages/core/src/rules/common/no-invalid-parameter-examples.ts]: Fixed, thanks

File: packages/core/src/config/minimal.ts (hunk 2 above)
Comment (IgorKarpiuk) [reply packages/core/src/config/minimal.ts:36]: @RomanHotsiy @adamaltman, oas 3.0 has a "no-invalid-media-type-examples" rule by default, but 3.1 doesn't have this rule by default. Do we need to add it to minimal and recommended for 3.1?, it's only for 3.0 now.

Comment (RomanHotsiy) [reply packages/core/src/config/minimal.ts:36]: Yes, let's add it.

File: packages/core/src/config/minimal.ts [hunk 3]
```diff
@@ -47,6 +47,10 @@ export default {
     'no-servers-empty-enum': 'error',
   },
   oas3_1Rules: {
+    'no-invalid-media-type-examples': {
+      severity: 'warn',
+      allowAdditionalProperties: false,
+    },
```
Comment (adamaltman) [inline packages/core/src/config/minimal.ts:53]:
```suggestion
    'no-invalid-media-type-examples': 'warn',
```
Won't `allowAdditionalProperties` be `false` by default?

**Review by tatomyr — APPROVED** [review]

File: packages/core/src/config/minimal.ts (hunk 3 above)
Comment (IgorKarpiuk) [reply packages/core/src/config/minimal.ts:53]: Yes, it is false by default, but do we need to turn it off in mininal? @adamaltman

**Pull request merged**